src/Calendar.Plugin.Sample/SampleApp/ViewModels/AdvancedPageViewModel.cs

```diff
@@ -129,5 +138,31 @@ private async Task ExecuteEventSelectedCommand(object item)
                 await App.Current.MainPage.DisplayAlert(title, message, "Ok");
             }
         }
+
+        private void ChangeShownUnit(int amountToAdd)
+        {
+            switch (CalendarLayout)
```

Comment:
I think most of this is excess logic. Could we not just have a ChangeShown method (the same as ChangeShownWeek, where we send the proper parameter for changing the month aswell?